### DISCUSSION TITLE
Avoid to signup duplicated users

### DISCUSSION THREAD

**Discussion by RamboGj** [supabase/supabase#5241] [Questions] [answered]
Hey guys, I'm having a problem when trying to avoid creating an account when an user's email already exists. It's returning that the user already exists even if it does not exist on supabase's table, instead of creating an account
![Screenshot_1](https://user-images.githubusercontent.com/64593022/151728439-6bac0a31-e48b-4d78-acfb-16d47eeb5524.png)
![Screenshot_2](https://user-images.githubusercontent.com/64593022/151728444-d38a03f7-fb01-485d-8e09-308c88ade752.png)
.

**Comment by RamboGj** [comment]
Instead of fall down at the last else, every email I write, whatever it is, is being considered as an existing email

**Comment by GaryAustin1** [comment, marked as answer]
Supabase select will return an object containing a data array.  You seem to be just passing the object back which will always have something...
```
const { data, error } = await supabase
  .from('cities')
  .select()
```
Maybe you want return data.length  so it is 0 if empty or 1 if the email exists.

**Reply by RamboGj** [reply]
you're right, once it always return an empty array when the email does not exist. How should I exactly write that?

**Reply by GaryAustin1** [reply]
You could do "return response.data.length" 
This does not deal with error that might occur though.
0 would test as false and 1 as true.

**Reply by RamboGj** [reply]
Thank you!
